crates/oxc_linter/src/rules/eslint/no_useless_concat.rs

```diff
@@ -0,0 +1,92 @@
+use oxc_ast::ast::Expression;
+use oxc_ast::AstKind;
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+use oxc_syntax::operator::BinaryOperator;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Default, Clone)]
+pub struct NoUselessConcat;
+
+fn no_useless_concat_diagnostic(span0: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("eslint(no-useless-concat): Unexpected string concatenation of literals.")
+        .with_help("Rewrite into one string literal")
+        .with_labels([span0.into()])
+}
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Disallow unnecessary concatenation of literals or template literals
+    ///
+    /// ### Why is this bad?
+    ///
+    /// It’s unnecessary to concatenate two strings together.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// var foo = "a" + "b";
+    /// ```
+    NoUselessConcat,
+    suspicious
```

Comment:
This rule produces useless code, which fits in our category of `correctness`. I'll test this in real codebases and see if I can change this to `correctness`.

Or if you wanna try out yourself: https://github.com/oxc-project/oxlint-ecosystem-ci

Reply:
It seems like people love to concatenate strings ...